File: hileusesimple/src/main/java/com/xiongtao/hileusesimple/app/MyApplication.java
package com.xiongtao.hileusesimple.app;

import android.app.Application;

import dagger.hilt.android.HiltAndroidApp;

/**
 * @Description:
 * @Author: xiongtao
 * @Date: 2023-02-22 16:38
 */
// hilt 基本上都要 用 Application来辅助
@HiltAndroidApp
public class MyApplication extends Application {

}

File: app/src/main/java/com/xiongtao/jetpack/MyApplication.kt
package com.xiongtao.jetpack

import android.app.Application
import androidx.lifecycle.ProcessLifecycleOwner
import com.xiongtao.jetpack.lifecycle.ApplicationObserver

/**
 * @Description:
 * @Author: 86188
 * @Date: 2023/2/6 17:29
 */
class MyApplication: Application() {
    override fun onCreate() {
        super.onCreate()
        ProcessLifecycleOwner.get().lifecycle.addObserver(ApplicationObserver())
    }
}